### PR TITLE
CORE-20089: Use ExceptionDetails in body when exception is caught

### PR DESCRIPTION
**Pull request by BenYip123**
- Update exceptions to use ExceptionDetails when an exception that inherits from HttpApiException is caught. This allows for a more detailed exception with separate fields for title, exception cause and reason.
- Update tests that assert certain exceptions
- Reduce duplicate code where there are multiple catch blocks

E2E Tests PR: https://github.com/corda/corda-e2e-tests/pull/544
QA Regpack Tests PR: https://github.com/corda/qa-regpack-tests/pull/1073

### CODE REVIEW

**Comment by BenYip123** [comment]
build e2e

File: components/virtual-node/virtual-node-info-read-service-rest-extensions/src/main/kotlin/net/corda/virtualnode/read/rest/extensions/VirtualNodeInfoRPCExtensions.kt [hunk 1]
```diff
@@ -4,6 +4,7 @@ import net.corda.crypto.core.ShortHash
 import net.corda.crypto.core.ShortHash.Companion.LENGTH
```
Comment (vkolomeyko) [inline components/virtual-node/virtual-node-info-read-service-rest-extensions/src/main/kotlin/net/corda/virtualnode/read/rest/extensions/VirtualNodeInfoRPCExtensions.kt:4]: While we are on it we might also rename this file to `VirtualNodeInfoRestExtensions.kt`

**Review by vkolomeyko — COMMENTED** [review]
Looks fine - one minor comment posted.

Also we need to re-sync the branch as PR is in a conflicted state at the moment.

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**